### PR TITLE
Add msplim_ns for box 0

### PR DESCRIPTION
**Pull request by theamirocohen**
requies changes from PR https://github.com/ARMmbed/mbed-os/pull/5321
and changes from https://github.com/ARMmbed/uvisor/pull/486 
and https://github.com/ARMmbed/uvisor/pull/496

### CODE REVIEW

**Comment by theamirocohen** [comment]
please review @danny4478 @alzix @romkuz01

File: api/src/uvisor-input.S [hunk 1]
```diff
@@ -114,6 +114,9 @@ uvisor_config:
     /* Debug driver pointer */
     .long __uvisor_debug_driver
 
+    /* Stack limit for MSPNS_LIM */
```
Comment (romkuz01) [inline api/src/uvisor-input.S]: Change MSPNS_LIM to MSPLIM_NS in the comment.

File: core/system/inc/linker.h [hunk 2]
```diff
@@ -121,6 +121,9 @@ typedef struct {
     UvisorLibHooks const * const lib_hooks;
 
     TUvisorDebugDriver const * const debug_driver;
+
+    /* Stack limit for MSPNS_LIM */
```
Comment (romkuz01) [inline core/system/inc/linker.h]: Change MSPNS_LIM to MSPLIM_NS in the comment.

File: api/src/uvisor-input.S (hunk 1 above)
Comment (theamirocohen) [reply api/src/uvisor-input.S]: fix

File: core/system/inc/linker.h (hunk 2 above)
Comment (theamirocohen) [reply core/system/inc/linker.h]: fix

File: api/src/uvisor-input.S [hunk 3]
```diff
@@ -114,6 +114,9 @@ uvisor_config:
     /* Debug driver pointer */
     .long __uvisor_debug_driver
 
+    /* Stack limit for MSPLIM_NS */
+    .long __uvisor_stack_limit
```
Comment (alzix) [inline api/src/uvisor-input.S]: IMHO the variable name is misleading as it holds not the uvisor stack limit but the public box stack limit.
Am i missing something?

File: core/system/inc/linker.h [hunk 4]
```diff
@@ -121,6 +121,9 @@ typedef struct {
     UvisorLibHooks const * const lib_hooks;
 
     TUvisorDebugDriver const * const debug_driver;
+
+    /* Stack limit for MSPLIM_NS */
+    uint32_t * stack_limit;
```
Comment (alzix) [inline core/system/inc/linker.h]: 1. same here what stack limit is it used for?
2. Additional question - what will we do with this new field on ARMv7-M platforms?
3. Please note the naming conventions here - members are named exactly as in the `uvisor-input.S` except `__uvisor_` prefix

File: api/src/uvisor-input.S [hunk 5]
```diff
@@ -114,6 +114,9 @@ uvisor_config:
...
+
```
Comment (alzix) [inline api/src/uvisor-input.S]: i see you are missing the assignment here as you have assigned the value to the `__uvisor_stack_limit` from the mbed-os linker script.
Please consider assigning __StackLimit__ symbol directly without intermediate variable and without modifying mbed-os sources.

File: api/src/uvisor-input.S (hunk 3 above)
Comment (theamirocohen) [reply api/src/uvisor-input.S]: fixed

**Comment by alzix** [comment]
@Patater - please approve

File: api/src/uvisor-input.S [hunk 6]
```diff
@@ -114,6 +114,9 @@ uvisor_config:
     /* Debug driver pointer */
     .long __uvisor_debug_driver
 
+    /* Stack limit for publix box MSPLIM_NS, accessed by public_box_stack_limit */
```
Comment (Patater) [inline api/src/uvisor-input.S]:
```diff
- publix
+ public
```

File: api/src/uvisor-input.S [hunk 7]
```diff
@@ -134,6 +137,9 @@ __uvisor_debug_driver:
     .long 0
     .long 0
 
+__uvisor_public_box_stack_limit:
+    .long __StackLimit
```
Comment (Patater) [inline api/src/uvisor-input.S]: Why do we need the public box stack limit twice? Why not just put `__StackLimit` where `__uvisor_public_box_stack_limit` is instead?

Comment (theamirocohen) [reply api/src/uvisor-input.S]: We feel that the var name __StackLimit is confusing while having 4 different stucks, and we wish to keep uvisor-input and linker.h variables the same

File: api/src/uvisor-input.S (hunk 6 above)
Comment (theamirocohen) [reply api/src/uvisor-input.S]: fixed

**Pull request merged**